cpp/src/quantiles/quantiles.cu

```diff
@@ -83,32 +101,36 @@ namespace{ //unknown
       thrust::sort(rmm::exec_policy(stream)->on(stream), dv, dv+n);
     }
 
-    QuantiledIndex qi = find_quantile_index(n, q);
+    QuantiledIndex qi = find_quantile_index(n, quant);
 
     switch( interpolation )
     {
     case GDF_QUANT_LINEAR:
-      cudaMemcpy(&hv[0], dv+qi.lower_bound, sizeof(T), cudaMemcpyDeviceToHost); //TODO: async with streams
-      cudaMemcpy(&hv[1], dv+qi.upper_bound, sizeof(T), cudaMemcpyDeviceToHost); //TODO: async with streams
-      //TODO: safe operation to avoid overflow/underflow
-      result = static_cast<RetT>(static_cast<RetT>(hv[0]) + qi.fraction*static_cast<RetT>(hv[1]-hv[0]));
+      singleMemcpy(hv[0], dv+qi.lower_bound, stream);
+      singleMemcpy(hv[1], dv+qi.upper_bound, stream);
+      cudf::interpolate::linear(
+          cudf::detail::unwrap(result), 

```

Comment:
wrapper types may come to here,
Maybe we can move `unwrap` inside of `cudf::interpolate::linear`.
operator*(double, wrapper) is not defined.

Reply:
But in general, usage of `unwrap` should be pushed down to the lowest level possible. The risk of doing it at higher levels like this is that `wrapper` types that you don't want might silently slip through, e.g., trying to interpolate `cudf::category`.

It's best to be as explicit as possible to avoid silent errors. In this case, that would mean specializing `cudf::interpolate::linear` for `cudf::timestamp`, e.g.,

```
template <typename T_out>
void linear(T_out& result, cudf::timestamp lhs, cudf::timestamp rhs, double frac)
{
    double dlhs = static_cast<double>(unwrap(lhs));
    double drhs = static_cast<double>(unwrap(rhs));
    double one_minus_frac = 1.0 - frac;
    result = static_cast<T_out>(one_minus_frac * dlhs + frac * drhs);
}
```